### PR TITLE
feat: add support for useragent Fluent filter

### PR DESCRIPTION
**Pull request by csabatuz-chess**
This PR adds support for using the https://github.com/bungoume/fluent-plugin-ua-parser Fluent plugin where Fluent plugins are applicable.

Depends on: https://github.com/kube-logging/fluentd-images/pull/88

### CODE REVIEW

File: pkg/sdk/logging/model/filter/useragent.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+//
+//		@type ua_parser
+//		@id test_useragent
+//	 key_name my_agent
+//	 delete_key true
+//	 out_key ua_fields
+//	 flatten true
+//
+// </filter>
+// ```
```
Comment (pepov) [inline pkg/sdk/logging/model/filter/useragent.go:84]: please follow multiline formatting pattern based on this example:
https://github.com/kube-logging/logging-operator/blob/b4191ebb4d856b64cc5b6600d02e22c357b4a586/pkg/sdk/logging/model/syslogng/output/s3.go#L25

Comment (csabatuz-chess) [reply pkg/sdk/logging/model/filter/useragent.go]: If yes, would you like me to do that for all the filters in a separate PR?

Comment (pepov) [reply pkg/sdk/logging/model/filter/useragent.go]: It seems we haven't fixed all the bad patterns yet. You don't have to fix all the other filters, but let's format this one properly.

Follow up: https://github.com/kube-logging/logging-operator/issues/1596

Comment (csabatuz-chess) [reply pkg/sdk/logging/model/filter/useragent.go]: Ok, I don't know if what I did is what you meant, but I did the best I could figure.

**Comment by csabatuz-chess** [comment]
I'm stuck a bit, can't really interpret the failing E2E test, or connect it to my changeset.

**Comment by OverOrion** [comment]
> I'm stuck a bit, can't really interpret the failing E2E test, or connect it to my changeset.

I reran the CI and it passed, seems like the `TestVolumeDrain_Downscale` might be flaky, sorry about that!

File: pkg/sdk/logging/model/filter/useragent.go (hunk 1 above)
Comment (pepov) [reply pkg/sdk/logging/model/filter/useragent.go]: thanks, I've made a few updates to the formatting, but seems to be good now!

**Review by pepov — APPROVED** [review]

**Comment by csabatuz-chess** [comment]
image
The <highlight xml> directives don't seem to render right in Markdown, that's why I left with ```yaml.

Not sure what's downstream, if this'll be straightened out in the docs, just checking in if this was intended.

**Comment by OverOrion** [comment]
> image The directives don't seem to render right in Markdown, that's why I left with ```yaml.
> Not sure what's downstream, if this'll be straightened out in the docs, just checking in if this was intended.

Yeah, that is alright, it is deliberate. The generated *.md files will be consumed by Hugo and it will render it properly (indentation and highlighting wise as well).

**Review by OverOrion — APPROVED** [review]

**Pull request merged**